DuckDB.NET.Data/DuckDBSchema.cs

```diff
@@ -50,6 +54,52 @@ internal static class DuckDBSchema
             }
         };
 
+    private static DataTable GetDataSourceInformation(string? serverVersion) =>
+        new(DbMetaDataCollectionNames.DataSourceInformation) {
+            Columns =
+            {
+                { DbMetaDataColumnNames.CompositeIdentifierSeparatorPattern, typeof(string) },
+                { DbMetaDataColumnNames.DataSourceProductName, typeof(string) },
+                { DbMetaDataColumnNames.DataSourceProductVersion, typeof(string) },
+                { DbMetaDataColumnNames.DataSourceProductVersionNormalized, typeof(string) },
+                { DbMetaDataColumnNames.GroupByBehavior, typeof(GroupByBehavior) },
+                { DbMetaDataColumnNames.IdentifierPattern, typeof(string) },
+                { DbMetaDataColumnNames.IdentifierCase, typeof(IdentifierCase) },
+                { DbMetaDataColumnNames.OrderByColumnsInSelect, typeof(bool) },
+                { DbMetaDataColumnNames.ParameterMarkerFormat, typeof(string) },
+                { DbMetaDataColumnNames.ParameterMarkerPattern, typeof(string) }, 
+                { DbMetaDataColumnNames.ParameterNameMaxLength, typeof(int) },
+                { DbMetaDataColumnNames.ParameterNamePattern, typeof(string) },
+                { DbMetaDataColumnNames.QuotedIdentifierPattern, typeof(string) },
+                { DbMetaDataColumnNames.QuotedIdentifierCase, typeof(IdentifierCase) },
+                { DbMetaDataColumnNames.StatementSeparatorPattern, typeof(string) },
+                { DbMetaDataColumnNames.StringLiteralPattern, typeof(string) },
+                { DbMetaDataColumnNames.SupportedJoinOperators, typeof(SupportedJoinOperators) }
+            },
+            Rows =
+            {
+                    { 
+                        "\\.", 
+                        "duckdb", 
+                        serverVersion, 
+                        serverVersion, 
+                        GroupByBehavior.Unrelated, 
+                        "(^\\[\\p{Lo}\\p{Lu}\\p{Ll}_@#][\\p{Lo}\\p{Lu}\\p{Ll}\\p{Nd}@$#_]*$)|(^\\[[^\\]\\0]|\\]\\]+\\]$)|(^\\\"[^\\\"\\0]|\\\"\\\"+\\\"$)", 
```

Comment:
Can you add comments explaining what these regexes do/mean and where they come from?

Reply:
Added comments.